tests/v2/actions/etcdsnapshot/creates.go

```diff
@@ -338,9 +338,13 @@ func CreateAndValidateSnapshotV2Prov(client *rancher.Client, podTemplate *corev1
 	snapshotToRestore := createdSnapshots[0].ID
 	createdSnapshotIDs := []string{}
 	// prioritize s3 snapshots over local.
+	s3Found := false
+
 	for _, snapshot := range createdSnapshots {
-		if strings.Contains(snapshot.ID, "-s3") {
+		store, ok := snapshot.Annotations["etcdsnapshot.rke.io/storage"]
+		if ok && store == "s3" {
 			snapshotToRestore = snapshot.ID
+			s3Found = true
```

Comment:
oh, we already have this here. in that case can we not add a flag for s3 and send it to the function?

Reply:
I prefer not to add to the parameters to the func, since we already have to get the objects in the poll anyways. There will be some form of redundancy either way, but with the current implementation at least we're following the practice of limiting parameters to a function.